rust/shalom_core/src/operation/parse.rs

```diff
@@ -121,13 +123,40 @@ fn parse_selection_set(
     selection
 }
 
+fn parse_operation_type(operation_type: ApolloOperationType) -> OperationType {
+    match operation_type {
+        ApolloOperationType::Query => OperationType::Query,
+        ApolloOperationType::Mutation => OperationType::Mutation,
+        ApolloOperationType::Subscription => OperationType::Subscription,
+    }
+}
+
 fn parse_operation(
     global_ctx: &SharedShalomGlobalContext,
     op: Node<apollo_compiler::executable::Operation>,
     name: String,
     file_path: PathBuf,
 ) -> SharedOpCtx {
-    let mut ctx = OperationContext::new(global_ctx.schema_ctx.clone(), file_path);
+    let query = op.to_string();
+    let mut ctx = OperationContext::new(
+        global_ctx.schema_ctx.clone(),
+        query,
+        file_path,
+        parse_operation_type(op.operation_type),
+    );
+    for variable in op.variables.iter() {
+        let name = variable.name.to_string();
+        let default_value = variable.default_value.as_ref().map(|v| v.to_string());
+        let is_optional = !variable.ty.is_non_null();
+        let ty_name = variable.ty.inner_named_type().to_string();
```

Comment:
again thats an enum. you should match it and panic with a `todo!` macro for something that is not scalar.